text/0000-all-builder-buildpackless.md

```diff
@@ -0,0 +1,76 @@
+# Make All Builders Buildpackless
+
+## Summary
+
+Builders that conglomerate buildpacks are unsustainable in the long run as
+there is a layer cap set in place by Docker of 127 layers. We should stop
+adding buildpack to our builder and instead only offer builders that contain
+the stack and lifecycle. 
+
+## Motivation
+
+With the current size of the `full` builder we are already seeing users running
+into issues with exceeding the layer limit of the buildpack when trying to add
+custom buildpacks to their builds. This will only be exacerbated going forward
+with the addition of new buildpacks into the Paketo project. Not only that the
+`full` builders are completely unsustainable in the near future with proposals
+such as the addition of APMs and the restructuring of existing buildpacks which
+will both increase the layer count further.
+
+Currently the purpose of the builders that include buildpacks is to allow first
+time users to quickly plug in a large number of applications into the build
+process and see what comes out. However using these loaded builders is not the
+optimal solution if you are running builds for production images and the
+builder images are large and bloated for most users who only have applications
+in a couple of language domains.
+
+This layer cap will only continue to become more of a problem and there is no
+clear workaround so unfortunately a drastic restructuring is required.
```

Comment:
I'm pretty sure you can safely squash buildpack layers, at least from the standpoint that buildapack A should not contain files that overlap with files from buildpack B or any other buildpack. Thus squashing them down should be safe.

A quick glance, see my other comment, it does look like we'd need a custom tool though because buildpack metadata in the labels would need to be adjusted, or we'd need proper support in `pack`.

If you could squash layers, you could have one per composite, which would lower the limit to the point where the max layer count would never be an issue.